File: WorldFeed/Server/src/WorldFeed.Common/Models/WebhookSubscription.cs
namespace WorldFeed.Common.Models
{
    using global::WorldFeed.Common.Public.Models.Interfaces;
    using global::WorldFeed.Common.Public.Models.Interfaces.DTO.Webhooks;
    
    public class WebhookSubscription : IWebhookSubscription
    {
        private readonly IWebhookSubscriptionDTO subscriptionDTO;

        public WebhookSubscription(IWebhookSubscriptionDTO subscriptionDTO)
        {
            this.subscriptionDTO = subscriptionDTO;
        }

        public string UserId => this.subscriptionDTO.UserId;
    }
}

File: WorldFeed/Client/src/app/core/Core/Models/WebhookSubscription.ts
import {IWebhookSubscription} from "../../Public/Models/Interfaces/IWebhookSubscription";
import {IWebhookSubscriptionDTO} from "../../Public/Models/Interfaces/DTO/Webhooks/IWebhookEnvironmentSubscriptionsDTO";

export class WebhookSubscription implements IWebhookSubscription {
  private readonly _subscriptionDTO: IWebhookSubscriptionDTO;

  constructor(subscriptionDTO: IWebhookSubscriptionDTO) {
    this._subscriptionDTO = subscriptionDTO;
  }

  get userId(): string {
    return this._subscriptionDTO.userId;
  }
}
